WeWatch/WeWatch/Service/Webservice.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Webservice.swift
+//  WeWatch
+//
+//  Created by Anton on 30/11/2024.
+//
+
+import Foundation
+
+enum AuthenticationError: Error {
+    case invalidCredentials
+    case invalidResponse
+    case decodingError
+    case custom (errorMessage: String)
+}
+
+struct LoginRequestBody: Encodable {
+    let apikey: String
+    let pin: String
+}
+
+struct LoginResponse: Codable {
+    let status: String
+    let data: LoginData?
+}
+
+struct LoginData: Codable {
+    let token: String?
+}
+
+enum  HttpMethod {
+    case get([URLQueryItem])
+    case post(Data?)
+    
+    var name: String {
+        switch self{
+        case .get:
+            return "GET"
+        case .post:
+            return "POST"
+        }
+    }
+}
+
+struct Resource<T: Codable> {
+    let url: URL
+    var method: HttpMethod = .get([])
+}
+
+class Webservice {
+    
+    func call <T: Codable>(_ resource: Resource<T>) async throws -> T {
+        var request = URLRequest(url: resource.url)
```

Comment:
Let’s agree to always check your code before pushing to avoid repeating the same issues. Code style: While style isn’t our main focus, it’s important to write clean, well-structured code from the start:
• Remove unnecessary blank lines.
• Add blank lines where they are needed (e.g., after the first line of a struct, enum, or class).
• Always include access modifiers.
• Explicitly specify types wherever possible.